pkg/utils/helper.go

```diff
@@ -59,24 +67,82 @@ func RunCommandOnHost(command string, arg ...string) (string, error) {
 	return string(out), nil
 }
 
-// RunCommandOnContainer runs a command on container system
-func RunCommandOnContainer(command string, arg ...string) (string, error) {
+// RunCommandOnContainerWithOutputStreams runs a command on container system and returns both the stdout and stderr output streams
+func RunCommandOnContainerWithOutputStreams(command string, arg ...string) (CommandOutputStreams, error) {
 	cmd := exec.Command(command, arg...)
 
+	var stdout bytes.Buffer
+	var stderr bytes.Buffer
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+
+	err := cmd.Run()
+	outputStreams := CommandOutputStreams{stdout.String(), stderr.String()}
+
+	if err != nil {
+		return outputStreams, fmt.Errorf("Fail to run command in container: %s", fmt.Sprint(err)+": "+stderr.String())
+	}
+
+	return outputStreams, nil
+}
+
+// RunCommandOnContainer  runs a command on container system and returns the stdout output stream
+func RunCommandOnContainer(command string, arg ...string) (string, error) {
+	outputStreams, err := RunCommandOnContainerWithOutputStreams(command, arg...)
+	return outputStreams.Stdout, err
+}
+
+// RunBackgroundCommand starts running a command on a container system in the background and returns its process ID
+func RunBackgroundCommand(command string, arg ...string) (int, error) {
+	cmd := exec.Command(command, arg...)
 	var out bytes.Buffer
 	var stderr bytes.Buffer
 	cmd.Stdout = &out
 	cmd.Stderr = &stderr
-	err := cmd.Run()
+	err := cmd.Start()
 	if err != nil {
-		return "", fmt.Errorf("Fail to run command in container: %s", fmt.Sprint(err)+": "+stderr.String())
+		return 0, fmt.Errorf("Fail to start background command in container: %s", fmt.Sprint(err)+": "+stderr.String())
 	}
+	return cmd.Process.Pid, nil
+}
 
-	return out.String(), nil
+// Finds and kills a process with a given process ID
+func KillProcess(pid int) error {
+	process, err := os.FindProcess(pid)
+	if err != nil {
+		return fmt.Errorf("Failed to find process with pid %d to kill: ", pid, fmt.Sprint(err))
+	}
+	if err := process.Kill(); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Tries to issue an HTTP GET request up to maxRetries times
+func GetUrlWithRetries(url string, maxRetries int) ([]byte, error) {

```

Comment:
Same stuff, less code :) 
```go
func GetUrlWithRetries(url string, maxRetries int) ([]byte, error) {
	retry := 1
	for {
		resp, err := http.Get(url)
		if err != nil {
			if retry == maxRetries {
				return nil, fmt.Errorf("max retries reached for request HTTP Get %s: %w", url, err)
			}
			time.Sleep(5 * time.Second)
		} else {
			defer resp.Body.Close()
			log.Printf("HTTP GET request succeeded for url %s", url)
			return ioutil.ReadAll(resp.Body)
		}
	}
}
```

Reply:
@arnaud-tincelin this looks good, just one quick question. Does `retry` increment on its own or should I add a `retry++` before `time.Sleep(5 * time.Second)`

Reply:
@SanyaKochhar you are right, I forgot a retry++. I admit I didn't test this code :)
Additionally, I don't know if the `log.Printf("HTTP GET request succeeded for url %s", url)` line if very useful, it's up to you to keep it or not :)

Why is this retry mechanism needed? Is it normal the HTTP Get fails from time to time?

Reply:
You're right, the extra log line made more sense with the old loop's error messaging but is redundant now. 

The retry mechanism is needed because it seems like there is sometimes a lag between when the port forwarding is started and the get request goes through, so each time the first request was failing